components/emoji_picker/emoji_picker_overlay.jsx

```diff
@@ -44,64 +45,54 @@ export default class EmojiPickerOverlay extends React.PureComponent {
         enableGifPicker: false,
     };
 
-    constructor(props) {
-        super(props);
-        this.state = {};
-    }
-
-    static emojiPickerPosition(props) {
-        const emojiTrigger = props.target();
-
-        if (typeof props.rightOffset !== 'undefined') {
-            return props.rightOffset;
+    emojiPickerPosition = memoize((emojiTrigger, rightOffset) => {
+        if (typeof rightOffset !== 'undefined') {
+            return rightOffset;
```

Comment:
does it make sense to memoize an argument that returns itself if present? wouldn't make more sense to do the checking in the calling code like:
```js
const result = typeof rightOffset !== 'undefined' ? rightOffset : emojiPickerPosition(emojiTrigger)
```
given that the memoize function only stores the last call, removing it would probably only help with the performance issues.

Reply:
0/5 on that, I'm ok with doing the check first